PLNSRVCE-1003: Pipeline service CI setup documentation

prietyc123 opened the pull request:
This changes aims to add the documentation for pipeline service CI setup

Signed-off-by: Priti Kumari [pkumari@redhat.com](mailto:pkumari@redhat.com)

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,20 @@
+# Steps to setup Hypershift Configuration
+
+1. Follow [ROSA Provisioning](https://github.com/openshift-pipelines/pipeline-service/tree/main/ci/docs/rosa_cluster_provision.md) documentation for getting an AWS cluster.
+2. Intall and Configure [Hypershift](https://github.com/openshift-pipelines/pipeline-service/tree/main/ci/docs/hypershift_setup.md) on it.
```
`Install`

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,20 @@
+
+# setup github app
+
+- Github app setup demo
```
Is this missing a link?

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,20 @@
+
+# Bitwarden Secret
+
+Few secrets where we create HypershiftDeployment Resources has been stored in bitwarden like ROSA cluster kubeconfig, AWS pull secret and key pair.
```
`Bitwarden` (appears multiple times)

Roming22 replied:
Still needs to be fixed

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,20 @@
+
+# Bitwarden Secret
+
+Few secrets where we create HypershiftDeployment Resources has been stored in bitwarden like ROSA cluster kubeconfig, AWS pull secret and key pair.
```
`have been stored`

Roming22 replied:
`Few secrets where we create HypershiftDeployment Resources` -> not clear to me, can you rephrase?

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,20 @@
+# Bitwarden Secret
+
+Few secrets where we create HypershiftDeployment Resources has been stored in bitwarden like ROSA cluster kubeconfig, AWS pull secret and key pair.
+If you do not have bitwarden account, try creating one by following [Bitwarden Account Setup](https://redhat.service-now.com/help?id=kb_article_view&sys_kb_id=28b1450e975fd9102f8d361e6253af65)
```
Missing `.` at the end of the sentence.

Roming22 commented on ci/docs/hypershift_setup.md:
```diff
@@ -0,0 +1,79 @@
+# Hypershift
+
+HyperShift is middleware for hosting OpenShift control planes at scale that solves for cost and time to provision, as well as portability across cloud service providers with strong separation of concerns between management and workloads. Clusters are fully compliant OpenShift Container Platform (OCP) clusters and are compatible with standard OCP and Kubernetes toolchains.
```
Maybe make the first mention of Hypershift a link to the repo or project page?

Roming22 commented on ci/docs/hypershift_setup.md:
````diff
@@ -0,0 +1,79 @@
+```
+2. Create MultiClusterEngine to enable Hypershift
+```yaml
+apiVersion: multicluster.openshift.io/v1
````
Instead of documenting the manifests, maybe they should leave in the repository (maybe in `ci/manifests/hypershift`)?

Roming22 commented on ci/docs/hypershift_setup.md:
````diff
@@ -0,0 +1,79 @@
+```
+4. Create an S3 bucket for Hypershift Operator with public-read
+```bash
+$BUCKET_NAME=plnsvcci-oidc
````
Instead of documenting the commands, maybe this can be put in a script (e.g. `ci/hack/hypershift_setup.sh`?)

The doc would become quite small and easy to follow, as it would explain what Hypershift, list the requirement(s) and then link to the script.

prietyc123 replied:
Agree with you +1

Roming22 commented on ci/docs/hypershift_setup.md:
```diff
@@ -0,0 +1,79 @@
+  --from-literal=aws_access_key_id='<aws key id>' \
+  --from-literal=aws_secret_access_key='<aws access key>' \
+  --from-literal=pullSecret='<Download your OpenShift Container Platform pull secret file from cloud.redhat.com/openshift/install/pull-secret by selecting Download pull secret.>' \
+  --from-file=ssh-publickey='/Users/username/.ssh/id_rsa.pub'
```
`/User` is a macos path. It's better to use `$HOME`.

Roming22 commented on ci/docs/hypershift_setup.md:
```diff
@@ -0,0 +1,79 @@
+spec:
+  installNamespace: open-cluster-management-agent-addon
+
+k -n local-cluster get ManagedClusterAddOn
```
`k` is not a standard command.

Roming22 commented on ci/docs/rosa_cluster_provision.md:
```diff
@@ -0,0 +1,34 @@
+# Red Hat OpenShift Service on AWS (ROSA) Cluster Provisioning
+
+`rosa` is a command line tool that simplifies the use of Red Hat OpenShift Service on AWS, also known as ROSA. This cluster is required to manage Hypershift operator for pipeline-service.
```
The two sentences are disjointed. The second sentence starts with `This cluster`, the cluster has not been referenced before, only the ROSA service.

Roming22 commented on ci/docs/rosa_cluster_provision.md:
````diff
@@ -0,0 +1,34 @@
+`rosa` is a command line tool that simplifies the use of Red Hat OpenShift Service on AWS, also known as ROSA. This cluster is required to manage Hypershift operator for pipeline-service.
+
+Execute the following commands to provision a ROSA cluster:
+```bash
````
Same comment as the hypershift doc, you should move commands into scripts.

prietyc123 commented:
I still need to add pac configuration steps for CI on this pr.

Roming22 commented on ci/hack/rosa_cluster_provision.sh:
```diff
@@ -0,0 +1,23 @@
+cd bootstrap-ks/rosa
+
+# Set the required environment variable before provisioning
+export ROSA_TOKEN=${ROSA_TOKEN:-"<your-rosa-token>"}
```
Question: Do we want to read those values from a yaml and upload that yaml to Bitwarden? Or are those values something we do not even want to share at the team level?

prietyc123 replied:
Yes we are not sharing those values at team level

Roming22 commented:
@prietyc123 Please resolve conversation when you submit an update that takes care of a comment.

prietyc123 commented on ci/hack/hypershift_setup.sh:
```diff
@@ -0,0 +1,65 @@
+    unset BW_ACCOUNT
+}
+
+echo "Get vault data from bitwarden"
```
> Please add the -d and -h options of those scripts as well.

could you please elaborate `-d` and `-h`. Where you want me to add this.

Roming22 replied:
See the parse_args function in other scripts.

prietyc123 commented:
> @prietyc123 Can you please address the comments so we can move forward?

@Roming22 Did I miss any comments which needs to be addressed? I think I addressed the comment you asked for. Could you please re-review.

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,20 @@
 # Continuous integration tests
 
-The OpenShift robot is used to trigger functional tests on the code base.
+Pipelines as Code triggers the pipelineRun in [.tekton](https://github.com/openshift-pipelines/pipeline-service/tree/main/.tekton) which executes the functional tests tasks for each PR, before merging it.
```
`https://github.com/openshift-pipelines/pipeline-service` still appeasr multiple times. Please fix all occurences.

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,28 @@
 # Continuous integration tests
 
-The OpenShift robot is used to trigger functional tests on the code base.
+Pipelines as Code triggers the PipelineRuns in [.tekton](../../.tekton) which executes the test tasks for each PR, before merging it.
```
`execute`

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,28 @@
+
+## Pre-requisite:
+
+- install [bw](https://bitwarden.com/help/cli/) cli
```
`Install`

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,28 @@
+## Pre-requisite:
+
+- install [bw](https://bitwarden.com/help/cli/) cli
+- install [aws](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) cli
```
`Install`.

Roming22 commented on ci/docs/rosa_cluster_provision.md:
```diff
@@ -0,0 +1,32 @@
+`rosa` is a command line tool that simplifies the use of Red Hat OpenShift Service on AWS, also known as ROSA.
+Visit official ROSA [documentation](https://access.redhat.com/products/red-hat-openshift-service-aws)
+
+We create a ROSA cluster on aws which manages Hypershift operator for pipeline-service.
```
`AWS` is still mispelled `aws` in various places.

Roming22 commented on ci/docs/rosa_cluster_provision.md:
```diff
@@ -0,0 +1,32 @@
+
+We create a ROSA cluster on aws which manages Hypershift operator for pipeline-service.
+
+# How do I provision?
```
There are still issues with headers.

Roming22 commented:
@prietyc123 Please go through all open comments.

prietyc123 commented:
> There are still issues with headers.

I think you are looking into previous change only. Correct me if I am wrong. The current change has the headers you suggested.
<img width="1360" alt="image" src="https://user-images.githubusercontent.com/51620925/214198798-164c85ba-8df6-4c82-9b97-261aa71b3483.png">

Even the bitwarden stuffs you pointed out has been changed in the current change. Not sure whether the current change is visible to you or not 🤔

prietyc123 commented:
@Roming22 I have resolved all your concern. Could you please re-review. Thanks

bnallapeta commented on ci/hack/rosa_cluster_provision.sh:
```diff
@@ -0,0 +1,99 @@
+
+provision_rosa_cluster() {
+  # This repo has the rosa provision and destroy cluster scripts
+  git clone git@github.com:stolostron/bootstrap-ks.git
```
Always a best practice to checkout a branch or tag or commit to avoid surprises in the future in case they change something in main in the future.

prietyc123 replied:
@bnallapeta Could you please verify, if I understood you correctly.

bnallapeta replied:
Looks good. :+1:

bnallapeta commented on ci/hack/hypershift_setup.sh:
```diff
@@ -0,0 +1,153 @@
+Usage: 
+    ${0##*/} ./hypershift_setup.sh [options]
+
+Intsall Hypershift operator on ROSA cluster
```
Intsall -> Install

bnallapeta commented on ci/docs/rosa_cluster_provision.md:
```diff
@@ -0,0 +1,32 @@
+
+## How to provision a cluster?
+
+### Pre-requisite:
```
Pre-requisite: -> Pre-requisites:

bnallapeta commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,23 @@
+
+## Setup GitHub app
+
+- Need to configure GitHub app for Pipelines as Code configuration into pipeline-service repository.
```
pipeline-service -> Pipeline Service
(multiple occurrences)

bnallapeta commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,23 @@
+
+Hypershift official [Documentation Guide](https://hypershift-docs.netlify.app/).
+
+### Pre-requisite:
```
Pre-requisite: -> Pre-requisites:

bnallapeta commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,23 @@
-TODO: add more information, link to the repository/folder controlling those tests.
+All the functional tests run on a hypershift AWS cluster.
+
+## How to configure Hypershift
```
Hypershift -> HyperShift
(multiple occurrences)

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,23 @@
+
+- Install [bw](https://bitwarden.com/help/cli/) cli
+- Install [aws](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) cli
+- Provision a ROSA cluster by following [documentation](https://github.com/openshift-pipelines/pipeline-service/tree/main/ci/docs/rosa_cluster_provision.md)
```
Use local link.

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,23 @@
+
+### Hypershift Setup
+
+You can configure hypershift on ROSA by running the [hypershift_setup.sh](https://github.com/openshift-pipelines/pipeline-service/tree/main/ci/hack/hypershift_setup.sh) script.
```
Use local link

Roming22 commented on ci/docs/rosa_cluster_provision.md:
```diff
@@ -0,0 +1,32 @@
+
+### Configure Cluster:
+
+You can configure a ROSA cluster using [rosa_cluster_provision.sh](https://github.com/openshift-pipelines/pipeline-service/tree/main/ci/hack/rosa_cluster_provision.sh) script.
```
Use local link

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,23 @@
+Pipelines as Code triggers the PipelineRuns in [.tekton](../../.tekton) which execute the test tasks for each PR, before merging it.
 
-TODO: add more information, link to the repository/folder controlling those tests.
+All the functional tests run on a hypershift AWS cluster.
```
hypershift -> HyperShift (multiple occurences)

Roming22 commented on ci/hack/hypershift_setup.sh:
```diff
@@ -0,0 +1,153 @@
+}
+
+prechecks() {
+    KUBECONFIG="${KUBECONFIG:-}"
```
You are not checking for many of the mandatory arguments.

Roming22 commented on ci/hack/rosa_cluster_provision.sh:
```diff
@@ -0,0 +1,99 @@
+  tmpdir=$(mktemp -d)
+  cd "$tmpdir"  
+}
+
```
You are not checking for any of the mandatory arguments.

prietyc123 replied:
Added

Roming22 commented on ci/docs/continuous_integation.md:
```diff
@@ -1,5 +1,28 @@
+
+## Bitwarden secrets
+
+We import our hypershift secrets like pull-secret, public and private key pair, compute kubeconfig and base domain url to bitwarden.
```
HyperShift

Roming22 commented on ci/hack/hypershift_setup.sh:
```diff
@@ -0,0 +1,182 @@
+
+install_hypershift() {
+  echo "HyperShift setup on ROSA cluster"
+  # Enable HyperShift and make rosa cluster a managed cluster, visit documentation https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html-single/multicluster_engine/index#hosted-control-planes-configure 
```
ROSA

Roming22 commented on ci/hack/hypershift_setup.sh:
```diff
@@ -0,0 +1,182 @@
+  done
+  echo "HyperShift operator successfully installed on the managed cluster"
+
+  # Create aws credential secret
```
AWS

Roming22 commented on ci/hack/rosa_cluster_provision.sh:
```diff
@@ -0,0 +1,130 @@
+}
+
+provision_rosa_cluster() {
+  # This repo has the rosa provision and destroy cluster scripts
```
ROSA

Roming22 reviewed (commented):
Still a few typos, but otherwise looks good.

prietyc123 commented:
> Still a few typos, but otherwise looks good.

Done.
